phase/get_kubeconfig.go

```diff
@@ -15,51 +16,108 @@ type GetKubeconfig struct {
 	APIAddress string
 }
 
-type kubeconfig struct {
-	Clusters []struct {
-		Cluster struct {
-			Server string `yaml:"server"`
-		} `yaml:"cluster"`
-	} `yaml:"clusters"`
-}
-
 // Title for the phase
 func (p *GetKubeconfig) Title() string {
 	return "Get admin kubeconfig"
 }
 
+var readKubeconfig = func(h *cluster.Host) (string, error) {
+	return h.Configurer.ReadFile(h, h.Configurer.KubeconfigPath(h, h.K0sDataDir()))
+}
+
+var k0sConfig = func(h *cluster.Host) (dig.Mapping, error) {
+	cfgContent, err := h.Configurer.ReadFile(h, h.Configurer.K0sConfigPath())
+	if err != nil {
+		return nil, fmt.Errorf("read k0s config from host: %w", err)
+	}
+
+	var cfg dig.Mapping
+	if err := yaml.Unmarshal([]byte(cfgContent), &cfg); err != nil {
+		return nil, fmt.Errorf("unmarshal k0s config: %w", err)
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("parse k0s config: %w", err)
+	}
+
+	return cfg, nil
+}
+
 func (p *GetKubeconfig) DryRun() error {
 	p.DryMsg(p.Config.Spec.Hosts.Controllers()[0], "get admin kubeconfig")
 	return nil
 }
 
 // Run the phase
 func (p *GetKubeconfig) Run() error {
-	h := p.Config.Spec.K0sLeader()
+	h := p.Config.Spec.Hosts.Controllers()[0]
+
+	cfg, err := k0sConfig(h)
+	if err != nil {
+		return err
+	}
 
-	output, err := h.ExecOutput(h.Configurer.K0sCmdf("kubeconfig admin"), exec.Sudo(h))
+	output, err := readKubeconfig(h)
```

Comment:
I am actually in favor of using `k0s kubeconfig` instead of accessing the admin config directly.

Reply:
Yep that can be done